tensorflow/lite/examples/label_image/label_image.cc

```diff
@@ -47,6 +50,44 @@ namespace label_image {
 
 double get_us(struct timeval t) { return (t.tv_sec * 1000000 + t.tv_usec); }
 
+using TfLiteDelegatePtr = tflite::Interpreter::TfLiteDelegatePtr;
+using TfLiteDelegatePtrMap = std::map<std::string, TfLiteDelegatePtr>;
+
+#if defined(__ANDROID__)
+Interpreter::TfLiteDelegatePtr CreateGPUDelegate(Settings* s) {
+  TfLiteGpuDelegateOptions options;
+  options.metadata = TfLiteGpuDelegateGetModelMetadata(s->model->GetModel());
+  if (s->allow_fp16)
+    options.compile_options.precision_loss_allowed = 1;
+  else
+    options.compile_options.precision_loss_allowed = 0;
+  options.compile_options.preferred_gl_object_type =
+      TFLITE_GL_OBJECT_TYPE_FASTEST;
+  options.compile_options.dynamic_batch_enabled = 0;
+
+  return evaluation::CreateGPUDelegate(s->model, &options);
+}
+#endif  // defined(__ANDROID__)
+
+TfLiteDelegatePtrMap GetDelegates(Settings* s) {
+  TfLiteDelegatePtrMap delegates;
+  if (s->gl_backend) {
+#if defined(__ANDROID__)
```

Comment:
We can remove these `#if defined(__ANDROID__)` to make the code cleaner. We can just check if returned value from `CreateGPUDelegate(s)` or `CreateNNAPIDelegate()` was nullptr using `if (!delegate)` and just avoid populating it into the delegates map.